aep/general/0159/aep.md.j2

```diff
@@ -1,5 +1,89 @@
+
+```
+GET /v1/publishers/-/books?filter=...
+```
+
+- The URI pattern **must** still be specified with `*` and permit the
+  collection to be specified; a URI pattern **must not** hard-code the `-`
+  character.
+- The method **must** explicitly document that this behavior is supported.
+- The resources provided in the response **must** use the canonical name of the
+  resource, with the actual parent collection identifiers (instead of `-`).
+- Services **may** support reading across collections on `List` requests
+  regardless of whether the identifiers of the child resources are guaranteed
+  to be unique. However, services **must not** support reading across
+  collections on `Get` requests if the child resources might have a collision.
+- Cross-parent requests **should not** support `order_by`. If they do, the
+  field **must** document that it is best effort. This is because cross-parent
+  requests introduce ambiguity around ordering, especially if there is
+  difficulty reaching a parent (see [unreachable resources]).
+
+**Important:** If listing across multiple collections introduces the
+possibility of partial failures due to unreachable parents (such as when
+listing across locations), the method **must** indicate this following the
+guidance in [unreachable resources].
+
+### Unique resource lookup
+
+Sometimes, a resource within a sub-collection has an identifier that is unique
+across parent collections. In this case, it may be useful to allow a
+[`Get`](./get) method to retrieve that resource without knowing which parent
+collection contains it. In such cases, APIs **may** allow users to specify the
+wildcard collection ID `-` (the hyphen or dash character) to represent any
+parent collection:
+
+```
+GET https://library.exampleapis.com/v1/publishers/-/books/{book}
+```
+
+- The URI pattern **must** still be specified with `*` and permit the
+  collection to be specified; a URI pattern **must not** hard-code the `-`
+  character.
+- The method **must** explicitly document that this behavior is supported.
+- The resource name in the response **must** use the canonical name of the
+  resource, with actual parent collection identifiers (instead of `-`). For
+  example, the request above returns a resource with a name like
+  `publishers/123/books/456`, _not_ `publishers/-/books/456`.
+- The resource ID **must** be unique within parent collections.
+
+### Reading across collections with different path patterns
+
+Sometimes, a resource may have multiple possible path patterns. This typically
+happens when a resource, or one of its ancestors, may have more than one
+possible parent resource (including having no parent).
+
+For example, `Book` might have the following path patterns:
+
+- `publishers/{publisher}/books/{book}` for books with a publisher
+- `books/{book}` for self-published books
+
+In this case, APIs **may** allow users to read across _all_ collections of
+books by using the `-` wildcard character:
+
+`GET /v1/-/books`
```

Comment:
as an exercise - how would we document support for this behavior via something like OpenAPI?

Will there be a separate `GET` method documented for the hard-coded path of `/v1/-/books`? So it's effectively a custom GET method?

Reply:
I think this would be a separate route, but I have to say that I really dislike this pattern.

Reply:
I don't love it either, but I don't have a better idea for how to support this very real use case.  Open to ideas!

Reply:
@hudlow says:  `-` looks like a literal URL; `*` looks like it might be a pattern rather than an actual URL literal.  So without the history and transcoding context, even in a vacuum, `-` (or `_`) might be preferable to `*`.

Reply:
(marking resolved due to resolution offline)